Easydict/NewApp/Utility/Extensions/QueryService+ConfigurableService/OpenAIService+ConfigurableService.swift

```diff
@@ -9,27 +9,65 @@
 import Foundation
 import SwiftUI
 
-@available(macOS 12.0, *)
+protocol EnumLocalizedStringConvertible {
+    var title: String { get }
+}
+
+enum OpenAIModels: String, CaseIterable, Identifiable {
+    var id: Self {
+        self
+    }
+
+    case gpt3_5_turbo = "gpt-3.5-turbo"
+    case gpt3_5_turbo_1106 = "gpt-3.5-turbo-1106"
+    case gpt4
+    case gpt4_32k = "gpt-4-32k"
+}
+
+extension OpenAIModels: EnumLocalizedStringConvertible {
+    var title: String {
+        rawValue
+    }
+}
+
+@available(macOS 13.0, *)
 extension EZOpenAIService: ConfigurableService {
     func configurationListItems() -> some View {
-        ServiceStringConfigurationSection(
-            textFieldTitleKey: "service.configuration.openai.api_key.header",
-            headerTitleKey: "service.configuration.openai.api_key.title",
-            key: .openAIAPIKey,
-            prompt: "service.configuration.openai.api_key.prompt",
-            footer: {
-                Text("service.configuration.openai.api_key.footer")
-            }
-        )
+        ServiceConfigurationSecretSectionView(service: self, observeKeys: [.openAIAPIKey]) {
+            ServiceConfigurationSecureInputCell(
+                textFieldTitleKey: "service.configuration.openai.api_key.title",
+                key: .openAIAPIKey,
+                placeholder: "service.configuration.openai.api_key.placeholder"
+            )
+            // endpoint
+            ServiceConfigurationInputCell(
+                textFieldTitleKey: "service.configuration.openai.endpoint.title",
+                key: .openAIEndPoint,
+                placeholder: "service.configuration.openai.endpoint.placeholder"
+            )
+            // model
+            ServiceConfigurationPickerCell(
+                titleKey: "service.configuration.openai.model.title",
+                key: .openAIModel,
+                values: OpenAIModels.allCases
```

Comment:
Picker is good, but we also need to support custom model, allow user to input non-openai model name, like gemini-pro, since user may use custom api that converting other LLM api to openai api, such as [one-api](https://github.com/songquanpeng/one-api).

<img width="848" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/bfe53cdc-7914-4f9c-9c9d-3b7d56531d28">

Reply:
But I checked some apps using the openAI service(Bob, Opencat), seems they don't offer the custom input option.

I have checked the one-api repo, but IMHO, since this is `OpenAI service`, do we need to support other kinds of models which not included in OpenAI?

Reply:
This is needed, there is a certain threshold for OpenAI payment, not all users can use the official OpenAI API, they may use other LLM APIs (e.g. [chatglm](https://chatglm.cn/)), or even build LLM locally ([ChatGLM3](https://github.com/THUDM/ChatGLM3)), and we may not have enough effort to support all LLM APIs https://github.com/tisfeng/Easydict/issues/254#issuecomment-1842952932 ,  so I hope that although its name is OpenAI Translate, it is actually a general type of translation that supports the OpenAI API format.

<img width="925" alt="image" src="https://github.com/tisfeng/Easydict/assets/25194972/091d0d3b-b05e-487c-b011-f59cb1813d91">

Reply:
OK, last question, could we add a new separate service to support this? Like `OneAPI service`, use could configure them separately.

Reply:
You reminded me that the name `OpenAI Translate` name may be misleading to users, making them think that it only supports OpenAI translation. 

Maybe we can add some guiding prompts on this settings page, just like the introduction in the README above, to let users know that it can support generic types of OpenAI API.

Reply:
> OK, last question, could we add a new separate service to support this? Like `OneAPI service`, use could configure them separately.

This is ok. 

Actually I thought about making OpenAI a base type of service that allows users to customize and add multiple different channels of OpenAI APIs, similar to one-api, but that would require refactoring the code, I want to put later.

Reply:
> You reminded me that the name OpenAI Translate name may be misleading to users, making them think that it only supports OpenAI translation.

Yes, this is what I'm concerned about. 

And compared with inut model name, users will easy to use with the Picker style.

Reply:
It's fine to add a OneAPI service, which frees up OpenAI translations.

Note that you may need to add some hints to guide the user, like the README introduction, as some users may not be familiar with one-api.

Reply:
OK, do we need to create a new issue to track this?

Reply:
Yes,  to do this in a new issue is better.